editor-integration/emacs/dune.el

```diff
@@ -426,6 +426,17 @@ For customization purposes, use `dune-mode-hook'."
   (compile (format "%s build @@runtest" dune-command))
   (dune-promote))
 
+;;;###autoload
+(defun dune-find-dominating ()
+  "Find dominating dune file."
+  (interactive)
+  (if-let* ((dune-file (locate-dominating-file "." "dune"))
+            (dune-file (concat dune-file "dune")))
```

Comment:
This would be less ambiguous as

```suggestion
  (if-let* ((dune-file-dir (locate-dominating-file "." "dune"))
            (dune-file (concat dune-file-dir "dune")))
```